src/npe2/_plugin_manager.py

```diff
@@ -237,7 +237,15 @@ def __init__(
         self.events = PluginManagerEvents(self)
         self._npe1_adapters: List[NPE1Adapter] = []
         self._command_menu_map: Dict[str, Dict[str, Dict[str, List[MenuCommand]]]] = (
-            defaultdict(dict)
+            # for each manifest, maps command IDs to menu IDs to list of MenuCommands
+            # belonging to that menu
+            # i.e. manifest -> command -> menu_id -> list[MenuCommand]
+            # we keep a list of MenuCommands even though we've already indexed by
+            # command ID **and** menu ID because it's possible to declare the
+            # same command in the same menu with different `when` and `group`
+            # fields. This seems an unlikely use case, but it is possible in
+            # the current schema.
```

Comment:
Love the comment! I started reading it and found the mapping (line 242) and was like ooh boy I've forgotten this already why didn't I review when she pushed 😂, and by the time I finished it I was like, boom, that's right, this is great! 😃